multiversx_sdk/abi/address_value.py

```diff
@@ -42,10 +42,26 @@ def _check_pub_key_length(self, pubkey: bytes) -> None:
             raise ValueError(f"public key (address) has invalid length: {len(pubkey)}")
 
     def set_payload(self, value: Any):
-        pubkey = bytes(value)
+        if isinstance(value, dict):
+            value = cast(Dict[str, str], value)
```

Comment:
If cast can be skipped and the linter is okay with it, let's skip it.

Reply:
The linter complains if we skip the `cast`.